opendcs-web-client-jetty/build.gradle

```diff
@@ -4,12 +4,18 @@ plugins {
     id "opendcs-rest-api.java-conventions"
     id "opendcs-rest-api.deps-conventions"
     id "opendcs-rest-api.publishing-conventions"
+    id "war"
 }
 
 dependencies {
-    implementation(project(":opendcs-web-client"))
+    implementation(libs.jetty.jsp)
     implementation(libs.jetty.server)
+    implementation(libs.javax.servlet.jsp)
+    implementation(libs.servlet.api)
+    implementation(libs.jetty.annotations)
     implementation(libs.jetty.webapp)
+    implementation(libs.jersey.container.servlet)
+    implementation(libs.tomcat.embed)
```

Comment:
Tomcat dependency in the jetty installer?

Reply:
When I don't have this library added, I receive the following error when starting jetty:

java.lang.ClassNotFoundException: org.apache.catalina.filters.HttpHeaderSecurityFilter
        at java.net.URLClassLoader.findClass(URLClassLoader.java:387)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
        at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:352)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
        at org.eclipse.jetty.webapp.WebAppClassLoader.loadClass(WebAppClassLoader.java:538)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
        at org.eclipse.jetty.util.Loader.loadClass(Loader.java:64)
        at org.eclipse.jetty.servlet.BaseHolder.doStart(BaseHolder.java:89)
        at org.eclipse.jetty.servlet.FilterHolder.doStart(FilterHolder.java:97)
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:73)
        at org.eclipse.jetty.servlet.ServletHandler.lambda$initialize$0(ServletHandler.java:749)
        at java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
        at java.util.stream.Streams$ConcatSpliterator.forEachRemaining(Streams.java:742)
        at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:647)
        at org.eclipse.jetty.servlet.ServletHandler.initialize(ServletHandler.java:774)
        at org.eclipse.jetty.servlet.ServletContextHandler.startContext(ServletContextHandler.java:379)
        at org.eclipse.jetty.webapp.WebAppContext.startWebapp(WebAppContext.java:1449)
        at org.eclipse.jetty.webapp.WebAppContext.startContext(WebAppContext.java:1414)
        at org.eclipse.jetty.server.handler.ContextHandler.doStart(ContextHandler.java:916)
        at org.eclipse.jetty.servlet.ServletContextHandler.doStart(ServletContextHandler.java:288)
        at org.eclipse.jetty.webapp.WebAppContext.doStart(WebAppContext.java:524)
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:73)
        at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:169)
        at org.eclipse.jetty.server.Server.start(Server.java:423)
        at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:110)
        at org.eclipse.jetty.server.handler.AbstractHandler.doStart(AbstractHandler.java:97)
        at org.eclipse.jetty.server.Server.doStart(Server.java:387)
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:73)
        at org.opendcs.odcswebclient.jetty.Start.main(Start.java:131)
2024-01-16 21:23:18.681:WARN:oejw.WebAppContext:main: Failed startup of context o.e.j.w.WebAppContext@31ef45e3{                                              OpenDCS Web Application,/odcsweb,file:///tmp/jetty-0_0_0_0-8083-opendcs-web-client-99_bugfix-lib-updates-SNAPSH                                              OT_war-_odcsweb-any-1401972714019795226/webapp/,UNAVAILABLE}{/home/opendcs/OPENDCS/odcs-web-client/bin/opendcs-                                              web-client-99.bugfix-lib-updates-SNAPSHOT.war}
javax.servlet.UnavailableException: Class loading error for holder httpHeaderSecurity==org.apache.catalina.filt                                              ers.HttpHeaderSecurityFilter@134593bf{inst=false,async=true,src=DESCRIPTOR:file:///tmp/jetty-0_0_0_0-8083-opend                                              cs-web-client-99_bugfix-lib-updates-SNAPSHOT_war-_odcsweb-any-1401972714019795226/webapp/WEB-INF/web.xml}
        at org.eclipse.jetty.servlet.BaseHolder.doStart(BaseHolder.java:96)
        at org.eclipse.jetty.servlet.FilterHolder.doStart(FilterHolder.java:97)
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:73)
        at org.eclipse.jetty.servlet.ServletHandler.lambda$initialize$0(ServletHandler.java:749)
        at java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
        at java.util.stream.Streams$ConcatSpliterator.forEachRemaining(Streams.java:742)
        at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:647)
        at org.eclipse.jetty.servlet.ServletHandler.initialize(ServletHandler.java:774)
        at org.eclipse.jetty.servlet.ServletContextHandler.startContext(ServletContextHandler.java:379)
        at org.eclipse.jetty.webapp.WebAppContext.startWebapp(WebAppContext.java:1449)
        at org.eclipse.jetty.webapp.WebAppContext.startContext(WebAppContext.java:1414)
        at org.eclipse.jetty.server.handler.ContextHandler.doStart(ContextHandler.java:916)
        at org.eclipse.jetty.servlet.ServletContextHandler.doStart(ServletContextHandler.java:288)
        at org.eclipse.jetty.webapp.WebAppContext.doStart(WebAppContext.java:524)
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:73)
        at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:169)
        at org.eclipse.jetty.server.Server.start(Server.java:423)
        at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:110)
        at org.eclipse.jetty.server.handler.AbstractHandler.doStart(AbstractHandler.java:97)
        at org.eclipse.jetty.server.Server.doStart(Server.java:387)
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:73)
        at org.opendcs.odcswebclient.jetty.Start.main(Start.java:131)

Reply:
```
javax.servlet.UnavailableException: Class loading error for holder httpHeaderSecurity==org.apache.catalina.filt ers.HttpHeaderSecurityFilter@134593bf{inst=false,async=true,src=DESCRIPTOR:file:///tmp/jetty-0_0_0_0-8083-opend cs-web-client-99_bugfix-lib-updates-SNAPSHOT_war-_odcsweb-any-1401972714019795226/webapp/WEB-INF/web.xml}
```

Something is creating a really bad web.xml, there shouldn't be any references to a org.apache.catalina.filters class in a Jetty system. Something is very wrong, but it's not solved by including Tomcat embedded at this level.

Reply:
That looks to be caused by the web.xml file: https://github.com/opendcs/rest_api/blob/main/opendcs-web-client/src/main/webapp/WEB-INF/web.xml#L184

Reply:
I was able to clean up the WAR file and remove tomcat embedded.